.github/workflows/publish-release.yml

```diff
@@ -0,0 +1,143 @@
+name: Publish Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: Release version tag
+        required: true
+        type: string
+        default: 'latest'
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Setup node version
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+        cache: 'npm'
+        cache-dependency-path: '**/package-lock.json'
+
+    - name: Setup NPM version
+      run: npm install -g npm@7
+
+    - name: Get NPM cache directory
+      id: npm-cache-dir
+      run: echo "::set-output name=dir::$(npm config get cache)"
+
+    - name: Restore NPM cache
+      id: npm-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.npm-cache-dir.outputs.dir }}
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Install node modules
+      run: npm ci --ignore-scripts
+
+    - name: Download prebuild PR artifacts
+      uses: dawidd6/action-download-artifact@d0f291cf39bd21965ea9c4c6e210fc355c3844ed
+      with:
+        workflow: pr-realm-js.yml
+        commit: ${{ github.sha }}
+        path: ${{ github.workspace }}
+        workflow_conclusion: completed
+        github_token: ${{ secrets.REALM_CI_PAT }}
+        name: realm-js-prebuilds
+
+    - name: Read version
+      id: get-version
+      run: |
+        pkgVersion=$(jq -r .version package.json)
+        echo "::set-output name=version::$pkgVersion"
+        prerelease=false
+        if [ '${{ inputs.tag }}' != latest ]; then prerelease=true; fi
+        echo "::set-output name=prerelease::$prerelease"
+      shell: bash
+
+    - name: Npm Publish realm-js-${{ steps.get-version.outputs.version }}
+      run: npm publish --tag '${{ inputs.tag}}' realm-js@${{ steps.get-version.outputs.version }}
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+    - name: Find Release PR
+      uses: juliangruber/find-pull-request-action@f9f7484f8237cf8485e5ab826e542ba5dd9e9c6e
+      id: find-pull-request
+      with:
+        branch: ${{ github.ref }}
```

Comment:
Curious how this step works.  How does it find the correct PR?  What is `github.ref` and how is it set?

Reply:
The idea is that you run the publish-release step on the branch that was made for the release PR. So github.ref is just the ref for that branch.

Reply:
When you choose "publish release" from the list of manually triggered actions, you get a dropdown choice of which branch to run it on. There you can select the release branch.

github.ref is set by github itself at the start of every workflow, and it's the git ref (= commit or tag or branch or whatever) that triggered the workflow

Reply:
Oh great, that sounds like a good workflow.  Thanks for the clarification